dags/als-kp-dataset-dag.py

```diff
@@ -219,49 +218,116 @@ def transform_data(data: Dict[str, Any], **context) -> List[Dict[str, Any]]:
         return transformed_items
 
     @task
-    def create_datasets(transformed_items: List[Dict[str, Any]], **context) -> str:
-        """Create Synapse datasets and collection.
+    def find_duplicated_datasets(
+        transformed_items: List[Dict[str, Any]], **context
+    ) -> List[Dict[str, Any]]:
+        """Flag duplicated new C-PATH datasets.
 
         This task:
-        1. Creates a new Dataset Collection with predefined columns
-        2. For each transformed item:
-           - Creates a new Dataset with the item's title and description
-           - Adds the Dataset to the collection
-        3. Stores the collection in Synapse
+        1. Retrieves the current C-PATH datasets.
+        2. Identifies and flags new datasets that are duplicates based on their 'title'.
 
-        Arguments:
-            transformed_items: List of transformed and validated items
-            **context: Airflow task context containing DAG parameters
+        Args:
+            transformed_items (List[Dict[str, Any]]):
+                A list of transformed and validated items from the previous task.
+
+        Raises:
+            ValueError: If any 'sameAs' or 'title' values are missing or empty after transformation.
 
         Returns:
-            str: The Synapse ID of the created and stored dataset collection
+            List[Dict[str, Any]]: A list of duplicate dataset items if there are any.
         """
         syn_hook = SynapseHook(context["params"]["synapse_conn_id"])
         synapse_client = syn_hook.client
 
-        columns = [
```

Comment:
these are removed because stakeholders do not want the pipeline to change dataset collection schema.